From 3c3ad734069ac69965119504c0ce46f9df05ab76 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Horia=20Geant=C4=83?= <horia.geanta@nxp.com>
Date: Tue, 17 Apr 2018 10:16:35 +0300
Subject: [PATCH 016/235] crypto: caam/qi2 - fix several error paths
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

-some error paths lack a proper error code
-in case of deferred probing, do not print an error message
-make error code checking stricter, i.e. err != 0 instead of err < 0

Signed-off-by: Horia GeantÄƒ <horia.geanta@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/crypto/caam/caamalg_qi2.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/caamalg_qi2.c b/drivers/crypto/caam/caamalg_qi2.c
index 1b9aee64240e..ff5b25a8fbc8 100644
--- a/drivers/crypto/caam/caamalg_qi2.c
+++ b/drivers/crypto/caam/caamalg_qi2.c
@@ -5037,6 +5037,7 @@ static int __cold dpaa2_dpseci_dpio_setup(struct dpaa2_caam_priv *priv)
 						     dev);
 		if (unlikely(!ppriv->store)) {
 			dev_err(dev, "dpaa2_io_store_create() failed\n");
+			err = -ENOMEM;
 			goto err;
 		}
 
@@ -5518,12 +5519,13 @@ static int dpaa2_caam_probe(struct fsl_mc_device *dpseci_dev)
 	priv->ppriv = alloc_percpu(*priv->ppriv);
 	if (!priv->ppriv) {
 		dev_err(dev, "alloc_percpu() failed\n");
+		err = -ENOMEM;
 		goto err_alloc_ppriv;
 	}
 
 	/* DPSECI initialization */
 	err = dpaa2_dpseci_setup(dpseci_dev);
-	if (err < 0) {
+	if (err) {
 		dev_err(dev, "dpaa2_dpseci_setup() failed\n");
 		goto err_dpseci_setup;
 	}
@@ -5531,7 +5533,8 @@ static int dpaa2_caam_probe(struct fsl_mc_device *dpseci_dev)
 	/* DPIO */
 	err = dpaa2_dpseci_dpio_setup(priv);
 	if (err) {
-		dev_err(dev, "dpaa2_dpseci_dpio_setup() failed\n");
+		if (err != -EPROBE_DEFER)
+			dev_err(dev, "dpaa2_dpseci_dpio_setup() failed\n");
 		goto err_dpio_setup;
 	}
 
-- 
2.17.1

